browser/src/control/Control.JSDialog.js

```diff
@@ -748,32 +748,48 @@ L.Control.JSDialog = L.Control.extend({
 			if (this.map)
 				this.map._progressBar.end();
 
-			this.createContainer(instance, instance.overlay ? instance.overlay: instance.containerParent);
+			const dialogDomParent = instance.overlay ? instance.overlay: instance.containerParent;
+			const documentFragment = new DocumentFragment(); // do not modify dom yet
+
+			this.createContainer(instance, documentFragment);
 			this.createDialog(instance);
-			this.addHandlers(instance);
 
 			// FIXME: remove this auto-bound instance so it will be clear what is passed
 			instance.updatePos = this.setPosition.bind(this, instance);
 
-			// Special case for nonModal dialogues. Core side doesn't send their initial coordinates. We need to center them.
-			if (instance.nonModal && !(instance.startX && instance.startY)) {
-				this.centerDialogPosition(instance);
-			} else {
-				instance.updatePos();
-			}
+			app.appendLayoutingTask(() => {
+				// dialog built - add to DOM now
+				const existingNode = dialogDomParent.querySelector('[id="' + instance.container.id + '"]');
```

Comment:
Any reason not to use `getElementById` here?

Reply:
I'm not sure if will be unique... (it comes form the core) so wants to search inside container we worked on